proto/wippersnapper/sensor/v1/sensor.proto

```diff
@@ -0,0 +1,57 @@
+  SENSOR_TYPE_AMBIENT_TEMPERATURE            = 13; /** Temperature of the air around a sensor, in degrees Celsius, "{value}°C". */
+  SENSOR_TYPE_OBJECT_TEMPERATURE             = 14; /** Temperature of the object a sensor is touching/pointed at, in degrees Celsius, "{value}°C".*/
+  SENSOR_TYPE_VOLTAGE                        = 15; /** Volts, "{value}V". */
+  SENSOR_TYPE_CURRENT                        = 16; /** Milliamps, "{value}mA". */
+  SENSOR_TYPE_COLOR                          = 17; /** Values are in 0..1.0 RGB channel luminosity and 32-bit RGBA format. "Color: {value}".*/
+  SENSOR_TYPE_RAW                            = 18; /** Sensor reads a value which is not defined by this list, "Raw Value: {value}".*/
+  SENSOR_TYPE_PM10_STD                       = 19; /** Standard Particulate Matter 1.0, in ppm, "{value}ppm". */
+  SENSOR_TYPE_PM25_STD                       = 20; /** Standard Particulate Matter 2.5, in ppm, "{value}ppm". */
+  SENSOR_TYPE_PM100_STD                      = 21; /** Standard Particulate Matter 100, in ppm, "{value}ppm". */
+  SENSOR_TYPE_PM10_ENV                       = 22; /** Environmental Particulate Matter 1.0, in ppm, "{value}ppm". */
+  SENSOR_TYPE_PM25_ENV                       = 23; /** Environmental Particulate Matter 2.5, in ppm, "{value}ppm". */
+  SENSOR_TYPE_PM100_ENV                      = 24; /** Environmental Particulate Matter 100, in ppm, "{value}ppm".*/
+  SENSOR_TYPE_CO2                            = 25; /** Measured CO2, in ppm, "{value}ppm". */
+  SENSOR_TYPE_GAS_RESISTANCE                 = 26; /** Proportional to the amount of VOC particles in the air, in Ohms, "{value}Ω". */
+  SENSOR_TYPE_ALTITUDE                       = 27; /** Values are in meters (m), "${$v} m". */
+  SENSOR_TYPE_LUX                            = 28; /** Light level, in lux, "Lux: {value}". */
+  SENSOR_TYPE_ECO2                           = 29; /** equivalent/estimated CO2 in ppm (estimated from some other measurement), "{value}ppm". */
+  SENSOR_TYPE_UNITLESS_PERCENT               = 30; /** Percentage, unit-less, "{value}%". */
+  SENSOR_TYPE_AMBIENT_TEMPERATURE_FAHRENHEIT = 31; /** Temperature of the air around a sensor, in degrees Fahrenheit, "{value}°F". */
+  SENSOR_TYPE_OBJECT_TEMPERATURE_FAHRENHEIT  = 32; /** Temperature of the object a sensor is touching/pointed at, in degrees Fahrenheit, "{value}°F".*/
+  SENSOR_TYPE_VOC_INDEX                      = 33; /** Values are an index from 1-500 with 100 being normal, "${$v} VOC".*/
+  SENSOR_TYPE_NOX_INDEX                      = 34; /** Values are an index from 1-500 with 100 being normal, "${$v} NOx".*/
+  SENSOR_TYPE_TVOC                           = 35; /** Values are in parts per billion (ppb), "${$v} ppb". */
+}
+
+/**
+* SensorEvent  is used to return the sensor's value and type.
+*/
+message SensorEvent {
+  SensorType type = 1; /** The sensor's type and corresponding SI unit */
+  oneof value {
+    float float_value = 2; /** The sensor's value as a float. */
```

Comment:
@tyeth  ☝️  Any thoughts on accel/gyro/magnetometer sensor support within `sensor.proto`?

Reply:
I think this is why we made this a `oneof` field even though we only have `float_value` right now. We knew we might need to support a couple more types of values, just weren't ready to flesh them out entirely yet.

@PetteriAimonen is there a good way to specify an xyz triplet as a single value in protobufs? Or do we just break it out into another Message with 3 float values?

Reply:
I think a submessage representing 3 floats would be most logical. It will also map nicely to Vector3D and similar types available in many vector math libraries.

Reply:
Yeah, it would be similar to the mapping within [Adafruit_Sensor](https://github.com/adafruit/Adafruit_Sensor/blob/master/Adafruit_Sensor.h#L87) which can hold a 3D vector array, XYZ components or roll/pitch/heading.

```
  union {
    float v[3]; ///< 3D vector elements
    struct {
      float x; ///< X component of vector
      float y; ///< Y component of vector
      float z; ///< Z component of vector
    };         ///< Struct for holding XYZ component
```

Reply:
Added in latest commit.